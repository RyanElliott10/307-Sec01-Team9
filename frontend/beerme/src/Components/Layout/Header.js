import React from "react";
import { Link } from "react-router-dom";

import * as Constants from "../../Utils/Constants";
import UserController from "../../Controllers/UserController";

export function Header() {
  const onMouseOver = event => {
    event.target.style.color = Constants.HOVER_ORANGE_COLOR;
  };

  const onMouseOut = (event, key) => {
    let color = null;
    switch (key) {
      case 1:
        color = Constants.ORANGE_COLOR;
        break;
      case 2:
        color = "#000";
        break;
      default:
        color = "#000";
        break;
    }
    event.target.style.color = color;
  };

  const onTitleClick = () => {
    console.log("onTitleClick");
  };

  const onExploreClick = () => {
    console.log("onExploreClick");
  };

  const onRecClick = () => {
    console.log("onRecClick");
  };

  return (
    <div style={headerStyle}>
      <Link to="/" style={titleStyle} className="beerme-homepage-clickable">
        <h1
          onClick={onTitleClick}
          onMouseEnter={event => onMouseOver(event)}
          onMouseOut={event => onMouseOut(event, 1)}
        >
          BeerMe
        </h1>
      </Link>
      <div
        style={{ display: "flex", flexDirection: "row", paddingRight: "30px" }}
      >
        <Link
          to="/explore"
          style={exploreStyle}
          className={"explore_header_link"}
        >
          <h6
            onClick={onExploreClick}
            onMouseEnter={event => onMouseOver(event)}
            onMouseOut={event => onMouseOut(event, 2)}
          >
            Explore
          </h6>
        </Link>
        <Link
          to="/recommended"
          style={recStyle}
          className={"recommended_header_link"}
        >
          <h6
            onClick={onRecClick}
            onMouseEnter={event => onMouseOver(event)}
            onMouseOut={event => onMouseOut(event, 2)}
          >
            Recommended
          </h6>
        </Link>
        <Link
          to={UserController.getCurrentUser() ? "/account" : "/account-entry"}
          style={recStyle}
          className={"account_account_entry_header_link"}
        >
          <h6
            onClick={onRecClick}
            onMouseEnter={event => onMouseOver(event)}
            onMouseOut={event => onMouseOut(event, 2)}
          >
            Account
          </h6>
        </Link>
      </div>
    </div>
  );
}

const headerStyle = {
  display: "flex",
  flex: 1,
  background: "#F4F4F4",
  padding: "10px",
  flexDirection: "row",
  justifyContent: "space-between",
  paddingLeft: "30px",
  alignItems: "center",
  marginBottom: "20px"
};

const titleStyle = {
  color: Constants.ORANGE_COLOR,
  textDecoration: "none",
  paddingRight: "10px",
  cursor: "pointer"
};

const exploreStyle = {
  textDecoration: "none",
  paddingRight: "10px",
  cursor: "pointer",
  color: "#000"
};

const recStyle = {
  textDecoration: "none",
  paddingLeft: "10px",
  cursor: "pointer",
  color: "#000"
};

export default Header;
