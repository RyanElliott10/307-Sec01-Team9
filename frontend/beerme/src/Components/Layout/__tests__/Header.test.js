import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { MemoryRouter as Router } from "react-router-dom";

import Header from "../Header";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <Header />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with clicks", () => {
  const ref = mount(
    <Router>
      <Header />
    </Router>
  );

  ref
    .find("#explore-header-clickable")
    .at(0)
    .simulate("click");

  ref
    .find("#recommended-header-clickable")
    .at(0)
    .simulate("click");

  ref
    .find("#account-header-clickable")
    .at(0)
    .simulate("click");

  ref
    .find("#beerme-header-clickable")
    .at(0)
    .simulate("click");
});
