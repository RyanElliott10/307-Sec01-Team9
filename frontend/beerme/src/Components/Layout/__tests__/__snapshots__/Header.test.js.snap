// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "#F4F4F4",
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": "20px",
      "padding": "10px",
      "paddingLeft": "30px",
    }
  }
>
  <a
    className="beerme-homepage-clickable"
    href="/"
    onClick={[Function]}
    style={
      Object {
        "color": "#CC752A",
        "cursor": "pointer",
        "paddingRight": "10px",
        "textDecoration": "none",
      }
    }
  >
    <h1
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
    >
      BeerMe
    </h1>
  </a>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "paddingRight": "30px",
      }
    }
  >
    <a
      className="explore_header_link"
      href="/explore"
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
          "cursor": "pointer",
          "paddingRight": "10px",
          "textDecoration": "none",
        }
      }
    >
      <h6
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseOut={[Function]}
      >
        Explore
      </h6>
    </a>
    <a
      className="recommended_header_link"
      href="/recommended"
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
          "cursor": "pointer",
          "paddingLeft": "10px",
          "textDecoration": "none",
        }
      }
    >
      <h6
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseOut={[Function]}
      >
        Recommended
      </h6>
    </a>
    <a
      className="account_account_entry_header_link"
      href="/account-entry"
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
          "cursor": "pointer",
          "paddingLeft": "10px",
          "textDecoration": "none",
        }
      }
    >
      <h6
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseOut={[Function]}
      >
        Account
      </h6>
    </a>
  </div>
</div>
`;
